### PR TITLE
AV-203152: Fix: Failure in creating Virtual service, PG and other pools if name of one of the pool exceeds maximum avi controller length limit

### PR DESCRIPTION
**Pull request by akshayhavile**
…

### CODE REVIEW

**Comment by akshayhavile** [comment]
build ako

File: internal/nodes/avi_model_l7_hostname_shard.go [hunk 1]
```diff
@@ -167,43 +180,68 @@ func (o *AviObjectGraph) BuildPoolPGPolicyForDedicatedVS(vsNode []*AviVsNode, na
...
+		}
 		if !lib.GetNoPGForSNI() || !isIngr {
+			isPoolNameLenExceedAviLimit = false
 			pool_ref := fmt.Sprintf("/api/pool?name=%s", poolNode.Name)
+
 			ratio := obj.weight
-			pgNode.Members = append(pgNode.Members, &avimodels.PoolGroupMember{PoolRef: &pool_ref, Ratio: &ratio})
-			if vsNode[0].CheckPGNameNChecksum(pgNode.Name, pgNode.GetCheckSum()) {
-				vsNode[0].ReplaceSniPGInSNINode(pgNode, key)
+			if !isPoolNameLenExceedAviLimit {
```
Comment (arihantg) [inline internal/nodes/avi_model_l7_hostname_shard.go:218]: This condition will always be true. Can you explain the logic?

File: internal/nodes/avi_model_tls_passthrough.go [hunk 2]
```diff
@@ -85,6 +85,7 @@ func (o *AviObjectGraph) BuildVSForPassthrough(vsName, namespace, hostname, key
 	return avi_vs_meta
 }
 
+// hndle this
```
Comment (arihantg) [inline internal/nodes/avi_model_tls_passthrough.go:88]: typo

File: internal/nodes/avi_model_l7_hostname_shard.go (hunk 1 above)
Comment (akshayhavile) [reply internal/nodes/avi_model_l7_hostname_shard.go:218]: This is the miss. I will correct it.

Comment (akshayhavile) [reply internal/nodes/avi_model_l7_hostname_shard.go:217]: addressed

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Review by arihantg — APPROVED** [review]
LGTM

File: internal/nodes/avi_model_l7_translator.go [hunk 3]
```diff
@@ -456,30 +488,45 @@ func (o *AviObjectGraph) BuildPolicyPGPoolsForSNI(vsNode []*AviVsNode, tlsNode *
 
 			buildPoolWithInfraSetting(key, poolNode, infraSetting)
 
+			if lib.CheckObjectNameLength(poolNode.Name, lib.Pool) {
+				isPoolNameLenExceedAviLimit = true
+			}
```
Comment (DixitAakash) [inline internal/nodes/avi_model_l7_translator.go:493]: @akshayhavile should we also be invoking CheckObjectNameLength for PG here?

Comment (DixitAakash) [reply internal/nodes/avi_model_l7_translator.go:493]: Got it!

**Review by DixitAakash — APPROVED** [review]

**Pull request merged**